launcher/instancemanager.go

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright (C) 2024 Renesas Electronics Corporation.
+// Copyright (C) 2024 EPAM Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package launcher
+
+import (
+	"errors"
+	"time"
+
+	"github.com/aoscloud/aos_common/aoserrors"
+	"github.com/aoscloud/aos_common/aostypes"
+	"github.com/aoscloud/aos_communicationmanager/config"
+	"github.com/aoscloud/aos_communicationmanager/utils/uidgidpool"
+	log "github.com/sirupsen/logrus"
+)
+
+/**********************************************************************************************************************
+* Consts
+**********************************************************************************************************************/
+
+const removePeriod = time.Hour * 24
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+type instanceManager struct {
+	config                 *config.Config
+	storage                Storage
+	storageStateProvider   StorageStateProvider
+	validateTTLStopChannel chan struct{}
+	uidPool                *uidgidpool.IdentifierPool
+	removeServiceChannel   <-chan string
+}
+
+/***********************************************************************************************************************
+ * Private
+ **********************************************************************************************************************/
+
+func newInstanceManager(config *config.Config, storage Storage, storageStateProvider StorageStateProvider, removeServiceChannel <-chan string) (im *instanceManager, err error) {
+	im = &instanceManager{
+		config:                 config,
+		storage:                storage,
+		storageStateProvider:   storageStateProvider,
+		validateTTLStopChannel: make(chan struct{}),
```

Comment:
in my opinion much better and I think according to golang design used context with cancel instead of channel